### ISSUE TITLE
Running `strip` on compile outputs removes the embedded source code 

### ISSUE THREAD

**Issue by roj1512** [denoland/deno#22556]
Here is how it can be reproduced:

```shell
$ rm -rf hello_world
$ echo 'console.log("Hello, world!");' > hello_world.ts
$ deno compile hello_world.ts
$ strip hello_world
$ ./hello_world
error: No archive found.
```

This is not a serious issue in reality, but users could easily run into it if they mistakenly run `strip` on the compile outputs. Also, some tools do this automatically iirc (e.g., [yay](https://github.com/Jguer/yay)).

**Comment by sigmaSd** [comment]
That's probably expected, also as of latest version, of deno, deno uses an already stripped version of itself, so there is no gain really if you strip it again

**Comment by roj1512** [comment]
@sigmaSd I know that is expected, but it is also better to not have this caused in case of potential user-side mistakes.

**Comment by lino-levan** [comment]
I don't believe this to be solvable without some silly workarounds that would probably affect performance (embedding the module graph as a binary blob in the text segment?). I don't think it's really worth the time and maintenance headache this would be to solve.

**Comment by lucacasonato** [comment]
This is expected for now. Run strip on the deno binary prior to calling `deno compile` the binary.

**Comment by roj1512** [comment]
Where can the Deno binary be found? Isn’t it downloaded on-demand?

**Comment by roj1512** [comment]
Also I just tried `strip ~/.deno/bin/deno`, and now the binary is unsuable.

**Comment by roj1512** [comment]
```
 $ deno                                                                                                             
[1]    22475 killed     deno
```

**Comment by sigmaSd** [comment]
A special deno binary called denort (that have unneeded stuff removed ) is downloaded from here I think https://github.com/denoland/deno/releases/download/v1.44.2/denort-aarch64-unknown-linux-gnu.zip when you compile with a specific version the first time

**Comment by roj1512** [comment]
Yes, the binary is still too big with that. That is the reason I tried stripping it.

**Comment by lino-levan** [comment]
You should strip the `denort` binary and manually passing that in to deno compile.

**Comment by roj1512** [comment]
Where is the `denort` downloaded to and how should I pass another `denort` path to `deno compile`??

**Comment by littledivy** [comment]
There is no need to strip `denort`; it is stripped by default.

**Comment by roj1512** [comment]
> FYI There are some experiments going on to allow code-signing for deno compile which should also allow stripping compiled binaries

@littledivy So shouldn’t this issue be reopened?

**Comment by dsherret** [comment]
Reopening. PR got merged accidentally https://github.com/denoland/deno/pull/32171
